jpa/org.eclipse.persistence.jpa/src/main/java/org/eclipse/persistence/internal/jpa/metamodel/ManagedTypeImpl.java

```diff
@@ -1329,6 +1330,18 @@ protected void initialize() { // Future: Check all is*Policy() calls
         }
     }
 
+    public void preinitaliseMappings(AbstractSession session) {
+        for (DatabaseMapping mapping : getDescriptor().getMappings()) {
+            try {
+                mapping.preInitialize(session);
+            } catch (NullPointerException npe) {
```

Comment:
Sorry, but `NullPointerException` handling in this context doesn't suite project coding style.
It looks like more some workaround.
Please add test there which is used as a descriptor and prevent any regression in the future.
Good candidate for a test is following Maven module/location.
https://github.com/eclipse-ee4j/eclipselink/tree/master/jpa/eclipselink.jpa.testapps/jpa.test.fetchgroups/src/test/java/org/eclipse/persistence/testing/tests/jpa/fetchgroups

Reply:
Do you have a suggestion on how this should be handled instead?

Reply:
Find where NPE happens and avoid this call by something like `if (someObject != null) ....`.

Reply:
well, the proper way is to figure out what is null and if it can be null - if yes adding a null check is appropriate and if not - which is more likely - something needs to be set by some other code on a different place